backend/src/db_managers/users.py

```diff
@@ -1,6 +1,13 @@
+from typing import Any
+
+from src.core.database.utils import get_by
 from src.db_managers.base import BaseDatabaseManager
 from src.models import User
 
 
 class UserManager(BaseDatabaseManager):
     model = User
+
+    async def get_by_email(self, filters: dict[str, Any]) -> User:
+        stmt = get_by(self.model, query=self.base_query, criteria=filters)
+        return await self.session.scalar(stmt)
```

Comment:
What if there's no user with provided email? Could it be User | None?

```suggestion
    async def get_by_email(self, filters: dict[str, Any]) -> User | None:
        stmt: Select = get_by(self.model, query=self.base_query, criteria=filters)
        return await self.session.scalar(stmt) # if there's no User with such email it will throw an error if I'm not mistaken
```

Reply:
Btw I don't see the reason in this method, since from your UserService you can directly use `get_by` method from `BaseDatabaseManager`